src/utils/formatters.ts

```diff
@@ -0,0 +1,50 @@
+import { isArray } from 'lodash'
+import { IMiddleware } from '../types/interfaces'
+
+/**
+ * Works only with first middleware argument and prints his value if it consists of a primitive value array or an object containing such arrays
+ * @param {string} middlewareName
+ * @param {IMiddleware} middleware object containing all necessary information about actual middleware
+ * @return { string } middleware's description
+ * */
+export const basicArrayFormatter = (middlewareName: string, middleware: IMiddleware): string => {
+	const permissionsResult = `${middlewareName}: `
+
+	if (!middleware.isUsed) {
+		return `${permissionsResult}NO`
+	}
+
+	const { value } = middleware.middlewareArguments[0]
+
+	if (isArray(value)) {
+		return `${permissionsResult} ${value}`
+	}
+
+	return `${permissionsResult}<ul>${Object.keys(value)
+		.map((key) => {
+			let result = ''
+
+			Object.values(value[key]).forEach((array) => {
+				if (isArray(array)) {
+					result += `<li>${key}${array.length > 0 ? `: [${array.join(', ')}]` : ''}</li>`
+				}
+			})
+
+			if (result === '' && value[key].allowedPermissions) {
```

Comment:
@kot-1999 I do not think that there should be any direct access to allowedPermissions specifically. this function should be generic function which will return whole first param of middleware function